### ISSUE TITLE
Exclude special filetypes from Include via Ignore section in spec.yaml

### ISSUE THREAD

**Issue by kevya-google** [abcxyz/abc#158]
### TL;DR

An idea was proposed to add a separate .specignore file for excluding special system files (like .DS_Store) or other private files that should not be included in template renders. Having a separate file for this would require template readers to have to go back and forth between spec.yaml and .specignore to understand how the render should behave. Instead, this could be included as a separate optional "ignore" section in the spec.yaml. Adding either a .specignore or an ignore section in spec.yaml would also support nice-to-haves such as a future IDE plugin that could detect and gray out such files.

It was decided to hold off on this feature until we have a customer who needs this functionality.

An example of this might look like:
```
apiVersion: 'cli.abcxyz.dev/v1alpha1'
kind: 'Template'
desc: 'Example template'

ignore: ['.DS_Store, '.bin', '.ssh']

inputs:
steps:
...
```

[proposal for this feature in Include Enhancements doc](https://docs.google.com/document/d/12iN2g1Idz6fW5ZqaokTmSFZf5kHXaWeewULWy2q5GfE/edit#heading=h.2wia9q1hlzad)

Idea from @sethvargo.

**Comment by sqin2019** [comment]
@drevell @kevya-google is this feature going to replace the [skip](https://github.com/abcxyz/abc/blob/0402ed8413f02e1069c2aec368eca208895918b1/templates/model/spec/v1alpha1/spec.go#L291) in include action, if not, what are the differences?

**Comment by sqin2019** [comment]
Pending: support ignore filenames (eg. if the ignore path starts with "/", then it matches the full path. Otherwise, it matches just the filename)
